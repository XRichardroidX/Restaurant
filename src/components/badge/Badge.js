import React from 'react';
import PropTypes from 'prop-types';

export default function Badge({ ...props }) {
  return (
    <div
      data-testid="badge"
      className={`rounded-lg min-w-fit uppercase ${props.twClasses}`}
    >
      {props.text}
    </div>
  );
}

Badge.defaultProps = {
  twClasses: '',
  text: '',
};

Badge.propTypes = {
  text: PropTypes.string,
  twClasses: PropTypes.string,
};
