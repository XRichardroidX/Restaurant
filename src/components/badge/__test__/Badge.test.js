import {
  render,
  screen,
  cleanup,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Badge from '../Badge';

describe('Badge', () => {
  afterEach(cleanup);

  it('renders correctly in the DOM', () => {
    const tree = renderer.create(<Badge />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with the given classNames', () => {
    render(<Badge twClasses="bg-red-500" />);

    const badge = screen.getByTestId('badge');
    const { classList } = badge;

    expect(classList).toContain('bg-red-500');
  });

  it('renders with the correct text', () => {
    render(<Badge text="My-Badge" />);

    const badge = screen.getByTestId('badge');

    expect(badge.textContent).toContain('My-Badge');
  });
});
