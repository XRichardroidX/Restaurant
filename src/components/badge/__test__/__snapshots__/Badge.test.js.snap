// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly in the DOM 1`] = `
<div
  className="rounded-lg min-w-fit uppercase "
  data-testid="badge"
/>
`;
